### PR TITLE
[path_provider] Documentation update for readme.md and example app

### PR DESCRIPTION
**Pull request by jpsiitonen**
Update for documentation on readme file:
- added table for platforms support for different directory types
- added short API usage examples for different directories

Example application improved, added downloads directory action. 

Fixes https://github.com/flutter/flutter/issues/97000

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/path_provider/path_provider/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.0.9
+* Updates documentation on README.md.
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider/CHANGELOG.md:2]: Blank line above.

File: packages/path_provider/path_provider/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+* Update to example application
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider/CHANGELOG.md:3]: "Updates example application."

File: packages/path_provider/path_provider/README.md [hunk 3]
```diff
@@ -2,28 +2,206 @@
...
+A Flutter plugin for finding commonly used locations on the filesystem. 
+Supports Android, iOS, Linux, macOS and Windows.
 Not all methods are supported on all platforms.
 
 ## Usage
 
 To use this plugin, add `path_provider` as a [dependency in your pubspec.yaml file](https://flutter.dev/docs/development/platform-integration/platform-channels).
 
-### Example
+Add the following import:
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider/README.md:13]: We don't need to explain the basics of Dart package use in the README.

File: packages/path_provider/path_provider/README.md [hunk 4]
```diff
@@ -2,28 +2,206 @@
...
+```dart
+import 'package:path_provider/path_provider.dart';
+```
+
+## Supported platforms and paths
+
+Methods support by platform:
 
+| Dir | Android | iOS | Linux | macOS | Windows |
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider/README.md:22]: Directory

File: packages/path_provider/path_provider/README.md [hunk 5]
```diff
@@ -2,28 +2,206 @@
...
+  try {
+    final Directory directory = await getTemporaryDirectory();
+    final String dirPath = directory.path;
+    return dirPath;
+  } catch (e) {
+    // HANDLE ERROR
+  }
+  return null;
+}
+```
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider/README.md:49]: This is way too much boilerplate to repeat for each item.

File: packages/path_provider/path_provider/README.md [hunk 6]
```diff
@@ -2,28 +2,206 @@
...
+| Application Library | ❌️ | ✔️ | ✔️ | ✔️ | ✔️ |
+| Application Documents | ✔️ | ✔️ | ✔️ | ✔️ | ✔️ |
+| External Storage | ✔️ | ❌ | ❌ | ❌️ | ❌️ |
+| External Cache Directories | ✔️ | ❌ | ❌ | ❌️ | ❌️ |
+| External Storage Directories | ✔️ | ❌ | ❌ | ❌️ | ❌️ |
+| Downloads | ❌ | ❌ | ✔️ | ✔️ | ✔️ |
+
+### Temporary Directory
+Path to the temporary directory on the device that is suitable for storing caches of downloaded 
+files.
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider/README.md:35]: I don't think we want to duplicate API docs in the README; that is asking for maintenance issues.

File: packages/path_provider/path_provider/README.md [hunk 7]
```diff
@@ -2,28 +2,206 @@
...
+  try {
+    final Directory? directory = await getDownloadsDirectory();
+    final String? dirPath = directory?.path;
+    return dirPath;
+  } catch (e) {
+    // HANDLE ERROR
+  }
+  return null;
+}
+```
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider/README.md:199]: I think we should remove all of these sections. They don't seem to add any value over the API docs, and the examples are entirely rote.

If we want an example, we could show a single mini-example that shows a few common examples (support dir, documents directory, temp).

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/path_provider/path_provider/CHANGELOG.md (hunk 2 above)
Comment (jpsiitonen) [reply packages/path_provider/path_provider/CHANGELOG.md]: Updated description

File: packages/path_provider/path_provider/README.md (hunk 7 above)
Comment (jpsiitonen) [reply packages/path_provider/path_provider/README.md]: dropped out boilerplate.

File: packages/path_provider/path_provider/README.md [hunk 8]
```diff
@@ -2,28 +2,42 @@
...
+## Supported platforms and paths
+
+Directories support by platform:
 
-### Usage in tests
+| Directory | Android | iOS | Linux | macOS | Windows |
+| :--- | :---: | :---: | :---: | :---: | :---: |
+| Temporary | ✔️ | ✔️ | ✔️ | ✔️ | ✔️ |
+| Application Support | ✔️ | ✔️ | ✔️ | ✔️ | ✔️ |
+| Application Library | ❌️ | ✔️ | ✔️ | ✔️ | ✔️ |
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider/README.md:30]: I missed this before. This is not supported on Linux or Windows.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
@bparrishMines for second review since I just have the one comment remaining.

File: packages/path_provider/path_provider/README.md (hunk 8 above)
Comment (jpsiitonen) [reply packages/path_provider/path_provider/README.md]: Fixed.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**